transports/p2p/p2psync/manager.go

```diff
@@ -712,15 +711,18 @@ func (sm *SyncManager) handleInvMsg(imsg *invMsg) {
 		blkHeight, err := sm.Services.Headers.GetHeightByHash(&invVects[lastBlock].Hash)
 		if err == nil {
 			peer.UpdateLastBlockHeight(blkHeight)
+			// we know about the block, no need to fetch it
+			return
 		}
 	}
 
 	if lastBlock != -1 {
-		sm.log.Info().Msgf("[Manager] handleInvMsg  lastConfirmedHeaderNode.hash  : %s", lastHeader.Hash)
-		sm.log.Info().Msgf("[Manager] handleInvMsg lastConfirmedHeaderNode.height : %d", lastHeader.Height)
+		locator := sm.Services.Headers.LatestHeaderLocator()
+		sm.log.Info().Msgf("[Manager] handleInvMsg tip hash  : %s", locator[0])
```

Comment:
Are you sure that `index out of range` will not occur?

Reply:
At the minimum we should at least have Genesis block in the database, and it should be returned from LatestHeaderLocator. If not, then something is very wrong and this shouldn't ever has happen.